common/models/user.js

```diff
@@ -668,6 +668,26 @@ module.exports = function(User) {
       next();
     });
 
+    // Delete old sessions once email is updated
+    UserModel.observe('before save', function(ctx, next) {
+      if (ctx.instance) {
+        var oldEmail = ctx.instance.email;
+      }
+      UserModel.observe('after save', function(ctx, next) {
```

Comment:
This will add a new observer after whenever a save operation is called. That won't work.

The correct solution is to store `oldEmail` in `ctx.hookState` instead of creating a local closure.

```
// in before save
ctx.hookState.oldEmail = ctx.instance.email

// in after save
var oldEmail = ctx.hookState.oldEmail
```

Having said that, I am afraid we don't have any easy way how to detect when a field was changed.

The only reliable solution is to query the database for the old version and compare it with the data being persisted.